Integrations/IronDefense/IronDefense.py

```diff
@@ -0,0 +1,369 @@
+
+    def _configure_session_auth(self, data):
+        self.logger.debug('Getting jwt...')
+
+        jwt = self._get_jwt(data)
+        if jwt:
+            # set the auth token if it exists
+            self.session.headers.update({'Authorization': 'Bearer ' + jwt})
+
+    def _http_request(self, method, uri, body='{}', headers={}, params={}, auth=None, files=None):
+        # Makes an API call with the given arguments
+        resp = self.session.request(
+            method,
+            self.base_url + uri,
+            data=body,
+            headers=headers,
+            verify=False,
+            params=params,
+            files=files,
+            timeout=self.request_timeout,
+            auth=auth,
+        )
+        if resp.status_code == 401:
+            if auth is not None:
+                # incorrect creds have been provided
+                return resp
+
+            # the session has expired so we need to log in again
+            self.logger.debug('Login required!')
+
+            username = self.credentials.get('identifier')
+            password = self.credentials.get('password')
+
+            # retry the original request with basic auth credentials
+            return self._http_request(method, uri,
+                                      body=body,
+                                      params=params,
+                                      files=files,
+                                      auth=(username, password))
+        elif resp.status_code >= 500:
+            self.logger.error('A server error has occurred. The response is: ' + json.dumps(resp.json()))
+
+        if auth is not None:
+            # persist the jwt
+            jwt = resp.headers.get('auth-token')
+            # save the jwt
+            if self.data_store.data_store_method is DemistoDataStore.DataStoreMethod.LAST_RUN:
+                # the context does not exist which means we are running 'fetch-incidents'
+                # save the jwt in last run so we can retrieve it later
+                self.data_store.set_last_run('IronDefense', {
+                    'Sessions': {
+                        'Host': self.host,
+                        'JWT': jwt
+                    }
+                })
+            elif self.data_store.data_store_method is DemistoDataStore.DataStoreMethod.CONTEXT:
+                context_entry = {
+                    'Host': self.host,
+                    'JWT': jwt
+                }
+                # save the jwt in the context so we can retrieve it later
+                self.data_store.set_context('IronDefense.Sessions(obj.Host===val.Host)', context_entry,
+                                            msg='Successfully logged in')
+
+        return resp
+
+    def _get_error_msg_from_response(self, resp):
+        err_msg = resp.json().get('msg')
+        if err_msg is None:
+            err_msg = resp.text
+        return err_msg
+
+    '''MAIN FUNCTIONS'''
+
+    def fetch_incidents(self):
```

Comment:
was this code tested (not using unit-testing)? if not, add a comment here to specify that